lib/index.js

```diff
@@ -361,6 +361,18 @@ class VideoAsset extends BaseAsset {
             hatch._failed_assets.push(this);
         }));
     }
+
+    to_metadata() {
+        const metadata = super.to_metadata();
+        if (!metadata.canonicalURI) {
+            metadata.canonicalURI = this._download_uri;
+        }
+        if (!metadata.matchingLinks.length || !metadata.matchingLinks[0]) {
```

Comment:
Why do you need to check if the first item is falsey? Will that ever be the case?

Reply:
yes, see the BaseAsset to_metadata(), it sets `"matchingLinks": [ this._canonical_uri ],` which for the video can be `[ undefined ] `.